src/index.js

```diff
@@ -1,23 +1,45 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import { Provider } from "react-redux";
-import { createStore } from "redux";
+import { applyMiddleware, combineReducers, createStore } from "redux";
+import thunk from "redux-thunk";
+import { composeWithDevTools } from "redux-devtools-extension";
 
 import "./scss/index.scss";
 
 import App from "./App";
 import rootReducer from "./reducers/root";
 import * as serviceWorker from "./serviceWorker";
 
-import { loginWithBakery } from "./juju";
+import { fetchModelList, loginWithBakery } from "./juju";
+import jujuReducers from "./juju/reducers";
 
-loginWithBakery();
+const reduxStore = createStore(
+  combineReducers({
+    root: rootReducer,
+    juju: jujuReducers
+  }),
+  composeWithDevTools(applyMiddleware(thunk))
+);
+
+async function connectAndListModels(reduxStore) {
+  try {
+    const conn = await loginWithBakery();
+    reduxStore.dispatch(fetchModelList(conn));
+  } catch (error) {
```

Comment:
Related comment below, but rather than wrapping `dispatch` in a try catch, you should probably do your error handing in your thunk.

Reply:
I suppose I could move the dispatch out of the try/catch but the try/catch is actually for the `await`'d function.